src/config/ConfigBuilder.ts

```diff
@@ -383,6 +402,26 @@ export class ConfigBuilder {
         }
     }
 
+    /**
+     * Statistics properties are deprecated and metrics will be used instead. This method will ensure metrics properties
+     * are set if statistics properties are set.
+     */
+    private static ensureMetricsConfiguredIfStatisticsConfigured(config: ClientConfig) {
+        if (config.hasOwnProperty('properties') &&
+            config.properties.hasOwnProperty('hazelcast.client.statistics.enabled') &&
+            (!config.hasOwnProperty('metrics') || !config.metrics.hasOwnProperty('enabled'))) {
+            throw new RangeError('hazelcast.client.statistics.enabled property is deprecated in version 5.1. ' +
```

Comment:
Why do you throw `RangeError` here? This seems more like an illegal argument error.

Reply:
Here we rethrow IllegalConfigurationError anyway: https://github.com/hazelcast/hazelcast-nodejs-client/blob/c77b9c8094132376ad18defee792b6c166089406/src/config/ConfigBuilder.ts#L58

RangeError can be thrown if user provides something illegal. Think of the whole config object as input. So, we throw RangeError in handleConfig.

Reply:
According to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RangeError it seems like the value in question needs to be related to some range.

Reply:
Yes we can change this error. However, to be consistent with other errors, should I go and check all validation errors and ensure they are not RangeError if there is not a range to talk about? WDYT?

Reply:
Since `RangeError` is used for all other configuration errors, it's OK to leave it as is.